File: sdk/samples/WCFSamples/TechnologySamples/Scenario/Router/CS/router/Contracts.cs
// ----------------------------------------------------------------------------
// Copyright (C) 2003-2005 Microsoft Corporation, All rights reserved.
// ----------------------------------------------------------------------------
namespace Microsoft.ServiceModel.Samples
{
    using System;
    using System.ServiceModel.Channels;
    using System.ServiceModel;

    [ServiceContract(SessionMode = SessionMode.Allowed)]
    public interface ISimplexDatagramRouter
    {
        [OperationContract(IsOneWay = true, Action = "*")]
        void ProcessMessage(Message message);
    }

    [ServiceContract(SessionMode = SessionMode.Allowed)]
    public interface IRequestReplyDatagramRouter
    {
        [OperationContract(IsOneWay = false, Action = "*", ReplyAction = "*")]
        Message ProcessMessage(Message message);
    }

    [ServiceContract(SessionMode=SessionMode.Required)]
    public interface ISimplexSessionRouter
    {
        [OperationContract(IsOneWay = true, Action = "*")]
        void ProcessMessage(Message message);
    }

    [ServiceContract(SessionMode=SessionMode.Required, CallbackContract = typeof(ISimplexSessionRouter))]
    public interface IDuplexSessionRouter
    {
        [OperationContract(IsOneWay = true, Action = "*")]
        void ProcessMessage(Message message);
    }
}

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/Router/VB/router/Contracts.vb
' Copyright (C) 2003-2005 Microsoft Corporation, All rights reserved.

Imports System
Imports System.ServiceModel.Channels
Imports System.ServiceModel

Namespace Microsoft.ServiceModel.Samples

    <ServiceContract(SessionMode:=SessionMode.Allowed)> _
    Public Interface ISimplexDatagramRouter

        <OperationContract(IsOneWay:=True, Action:="*")> _
        Sub ProcessMessage(ByVal message As Message)

    End Interface

    <ServiceContract(SessionMode:=SessionMode.Allowed)> _
    Public Interface IRequestReplyDatagramRouter

        <OperationContract(IsOneWay:=False, Action:="*", ReplyAction:="*")> _
        Function ProcessMessage(ByVal message As Message) As Message

    End Interface

    <ServiceContract(SessionMode:=SessionMode.Required)> _
    Public Interface ISimplexSessionRouter

        <OperationContract(IsOneWay:=True, Action:="*")> _
        Sub ProcessMessage(ByVal message As Message)

    End Interface

    <ServiceContract(SessionMode:=SessionMode.Required, CallbackContract:=GetType(ISimplexSessionRouter))> _
    Public Interface IDuplexSessionRouter

        <OperationContract(IsOneWay:=True, Action:="*")> _
        Sub ProcessMessage(ByVal message As Message)

    End Interface

End Namespace
